go/api/v1alpha2/agent_types.go

```diff
@@ -85,6 +102,8 @@ type DeclarativeAgentSpec struct {
 
 	// +optional
 	Deployment *DeclarativeDeploymentSpec `json:"deployment,omitempty"`
+
+	ExecuteCode *bool `json:"executeCode,omitempty"`
```

Comment:
Can we be more specific about this field name, specifically this executor will execute code that is returned from the API directly, not files or anything like that. Also, maybe this should be an struct as we may want to support multiple executors in the future?